### PR TITLE
Timestamp module

### LINKED ISSUE #45: Timestamp module
Implement new Module -> Timestamp similar to Substrate's (https://docs.rs/pallet-timestamp/2.0.0-rc5/pallet_timestamp/)

The module must implement the following functions:

**`set`** - Sets the current time. When setting the new time, it must be greater than the last one (set into storage) with at least a `MinimumPeriod`. When setting the time,  it should set 2 key-values.

- The first one should have the following key: `{scale("timestamp")}{scale("now")}` and the value should be `{scale(now)}` where `now` is a timestamp number.
- The second one should have the following key: `{scale("timestamp")}{scale("didupdate")}` and the  value should be `{scale(bool)}` where bool is  `true`.

**`get`** - Gets the current time that was set. If this function is called prior to setting the timestamp, it will return the timestamp of the previous block.
Once called, the function must get the value set in the Storage at key `{scale("timestamp")}{scale("now")}`

**`MINIMUM_PERIOD`**  - Should be a constant defined in the AS code similar to how we are defining the `CORE_VERSION` constants. With the time we might think of a way of reading those from an `env` file or some kind of config, but in the beginning, we might hardcode it directly in the `Module` folder. The value should be in `u64` format and should hold the `milliseconds between slots` f.e 5000 -> 5 sec "block time"

**Inherents** - The module must be able to `create` and `check` inherents produced by this module

**Integration Tests** - New API tests should be added for testing the functionality

### PR DESCRIPTION
**Pull request by dastanbeksamatov**
Closes #45 
### Timestamp module implementation
- Add Timestamp class with get, set static methods
- Add integration tests in rust
- Add Timestamp module in yarn workspace
- Add ProvideInherent methods for the Timestamp

### CODE REVIEW

File: runtime/packages/modules/timestamp/assembly/index.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// The entry file of your WebAssembly module.
```
Comment (Daniel-K-Ivanov) [inline runtime/packages/modules/timestamp/assembly/index.ts:1]: you do not need this comment

File: runtime/packages/modules/timestamp/package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "@as-substrate/timestamp-module",
+  "version": "1.0.0",
+  "description": "Modules used in AS-Based Substrate Runtimes",
+  "author": "",
+  "license": "Apache-2.0",
+  "scripts": {},
+  "dependencies": {
+    "@as-substrate/balances-module": "1.0.0",
```
Comment (Daniel-K-Ivanov) [inline runtime/packages/modules/timestamp/package.json:9]: i think that this dependency is unnecessary

File: sandbox/tests/timestamp-module_test.rs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    let mut now = vec![];
+    let mut did_update = vec![];
+
+    now.extend(b"timestamp".to_vec().encode());
+    did_update.extend(b"timestamp".to_vec().encode());
+    now.extend(b"now".to_vec().encode());
+    did_update.extend(b"didupdate".to_vec().encode());
+
+    let init_value: u64 = 42;
+    ext.set_storage(now.clone(), init_value.encode());
```
Comment (Daniel-K-Ivanov) [inline sandbox/tests/timestamp-module_test.rs:53]: why are you setting the timestamp here? My thinking is that the value should be set in the AS runtime right?

Comment (dastanbeksamatov) [reply sandbox/tests/timestamp-module_test.rs:53]: I think setting the value in AS runtime and testing it again with get would introduce dependency to the `Timestamp.set()`. This approach is similar to the Rust implementation, but they have a special method in Timestamp `set_timestamp()` which is used only for testing.

File: runtime/packages/modules/timestamp/assembly/timestamp.ts [hunk 4]
```diff
@@ -6,12 +6,17 @@ import { UInt64, Bool, ByteArray } from 'as-scale-codec';
 export class Timestamp{
 
     /**
-     * Necessary constants for timestamp
+     * Minimum period between timestamps
      */
     public static readonly MINIMUM_PERIOD: u64 = 5;
+    /**
+     * Scale encoded key {scale("timestamp")}{scale("now")} 
+     * Scale encoded key {scale("timestamp")}{scale("didupdate")}
```
Comment (Daniel-K-Ivanov) [inline runtime/packages/modules/timestamp/assembly/timestamp.ts:14]: Can you move this comment above the variable directly

File: sandbox/tests/timestamp-module_test.rs (hunk 3 above)
Comment (Daniel-K-Ivanov) [reply sandbox/tests/timestamp-module_test.rs:53]: oh yeah you are right. You should set it using rust

**Pull request merged**